jetty-server/src/main/java/org/eclipse/jetty/server/HttpInput.java

```diff
@@ -130,9 +130,30 @@ public void addInterceptor(Interceptor interceptor)
         }
     }
 
-    private int get(Content content, byte[] bytes, int offset, int length)
+    private int get(Content content, ByteBuffer des)
     {
-        int consumed = content.get(bytes, offset, length);
+        int consumed;
+        if (des.hasArray())
```

Comment:
May I delete the des.hasArray branch?  The else branch can support same functionality.